tests/conftest.py

```diff
@@ -103,7 +103,7 @@ def tmpdir(request, tmpdir):
     # This should prevent us from needing a multiple gigabyte temporary
     # directory while running the tests.
     if not request.config.getoption("--keep-tmpdir"):
```

Comment:
A comment might have been a good idea here, to explain the use of `shutil.rmtree` (to prevent a possible accidental revert in a distant future)

Reply:
I included the comments in a follow-up PR #7577